site/tests/integration/login-flow.test.ts

```diff
@@ -0,0 +1,201 @@
+  );
+
+  await verificationCodeInput.fill(
+    await readValueFromRecentDummyEmail("Email verification code: "),
+  );
+  await verificationCodeInput.press("Enter");
+
+  const accountDropdownButton = page.locator(accountDropdownButtonSelector);
+  await expect(accountDropdownButton).toBeVisible();
+
+  await accountDropdownButton.click();
+  await expect(accountDropdownButton).toHaveText("A");
+
+  const accountDropdownPopover = page.locator(
+    '[data-testid="account-dropdown-popover"]',
+  );
+
+  await expect(accountDropdownPopover).toBeVisible();
+  await expect(accountDropdownPopover).toContainText("Alice@alice");
+
+  for (const [name, href] of [
+    ["Dashboard", "/dashboard"],
+    ["Your Profile", "/@alice"],
+    ["API Keys", "/settings/api-keys"],
+  ] as const) {
+    const item = accountDropdownPopover.locator(`a:has-text("${name}")`);
+    await expect(item).toBeVisible();
+    await expect(item).toHaveAttribute("href", href);
+  }
+
+  const logOutItem = accountDropdownPopover.locator(`span:has-text("Log Out")`);
+  await expect(logOutItem).toBeVisible();
+  await expect(logOutItem).toHaveAttribute("href", "");
+
+  await logOutItem.click();
+
+  await expect(accountDropdownButton).toBeHidden();
+
+  await openLoginModal({ page, isMobile });
+});
+
+test("login works for an existing user (via magic link)", async ({
+  browserName,
+  context,
+  isMobile,
+  page,
+}) => {
+  test.skip(
+    browserName === "webkit",
+    "https://app.asana.com/0/1202542409311090/1202652399221616",
+  );
+
+  await execa("yarn", ["exe", "scripts/seed-db.ts"]);
+
+  await page.goto("/");
+  const loginModal = await openLoginModal({ page, isMobile });
+
+  const emailInput = loginModal.locator(emailInputSelector);
+
+  await emailInput.fill("alice@example.com");
+  await emailInput.press("Enter");
+
+  await expect(loginModal.locator("text=Check your inbox")).toBeVisible();
+
+  const magicLink = await readValueFromRecentDummyEmail("Magic Link: ");
+
+  const page2 = await context.newPage();
+  await page2.goto(magicLink);
+
+  const accountDropdownButton2 = page2.locator(accountDropdownButtonSelector);
+  await expect(accountDropdownButton2).toBeVisible();
+  await expect(accountDropdownButton2).toHaveText("A");
+
+  await page.reload();
+  const accountDropdownButton = page.locator(accountDropdownButtonSelector);
+  await expect(accountDropdownButton).toHaveText("A");
+  await accountDropdownButton.click();
+
+  const accountDropdownPopover = page.locator(
+    '[data-testid="account-dropdown-popover"]',
+  );
+
+  await expect(accountDropdownPopover).toBeVisible();
+  await expect(accountDropdownPopover).toContainText("Alice@alice");
+  await accountDropdownPopover.locator(`span:has-text("Log Out")`).click();
+
+  await openLoginModal({ page, isMobile });
+
+  await page2.reload();
+  await openLoginModal({ page: page2, isMobile });
+});
+
+test("login modal screen 1 correctly handles interactions", async ({
```

Comment:
Do we have a "login modal screen 2"?

Reply:
yep, it's where we ask to enter verification code